pkg/dockerfile/generator.go

```diff
@@ -130,10 +136,61 @@ func (g *Generator) baseImage() (string, error) {
 	return "python:" + g.Config.Build.PythonVersion, nil
 }
 
-func (g *Generator) preamble() string {
-	return `ENV DEBIAN_FRONTEND=noninteractive
+func (g *Generator) preamble() (string, error) {
+	preamble := `ENV DEBIAN_FRONTEND=noninteractive
 ENV PYTHONUNBUFFERED=1
 ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/x86_64-linux-gnu:/usr/local/nvidia/lib64:/usr/local/nvidia/bin`
+
+	envVarLines, err := g.environmentVariables()
+	if len(envVarLines) > 0 {
+		preamble += "\n" + strings.Join(envVarLines, "\n")
+	}
+
+	return preamble, err
+}
+
+func (g *Generator) environmentVariables() ([]string, error) {
+	// build.environment: list of strings formatted like `KEY=<VALUE>`
+	environmentVariables := g.Config.Build.Environment
+
+	regexpVariableName := regexp.MustCompile("^[A-Za-z_][A-Za-z0-9_]*$")
+
+	var envVarKeys []string
```

Comment:
What's the intention behind this? So it's ordered correctly?

Reply:
That's right! (Though maybe there's a more idiomatic way to implement it in Go?) 

The order can be significant since environment variables can refer to each other. (As normal with Docker/Dockerfiles.)